### PR TITLE
[DEST] add DEST model

### PR DESCRIPTION
**Pull request by zhby99**
## Motivation

We are from NVIDIA and we have developed a simplified and inference-efficient transformer for dense prediction tasks. The method is based on SegFormer with hardware-friendly design choices, resulting in better accuracy and over 2x reduction in inference speed as compared to the baseline. We believe this model would be of particular interests to those who want to deploy an efficient vision transformer for production, and it is easily adaptable to other tasks. Therefore, we would like to contribute our method to mmsegmentation in order to benefit a larger audience.

The paper was accepted to [Transformer for Vision workshop](https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fsites.google.com%2Fview%2Ft4v-cvpr22%2Fpapers%3Fauthuser%3D0&data=05%7C01%7Cboyinz%40nvidia.com%7Cbf078d69821449d1f4c908dab5e8c7da%7C43083d15727340c1b7db39efd9ccc17a%7C0%7C0%7C638022308636438546%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=XtSgPQrbVgHxt5L9XkXF%2BGWvc95haB3kKPcHnsVIF3M%3D&reserved=0) at CVPR 2022, here below are some resource links:
Paper [https://arxiv.org/pdf/2204.13791.pdf](https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Farxiv.org%2Fpdf%2F2204.13791.pdf&data=05%7C01%7Cboyinz%40nvidia.com%7Cbf078d69821449d1f4c908dab5e8c7da%7C43083d15727340c1b7db39efd9ccc17a%7C0%7C0%7C638022308636438546%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=X%2FCVoa6PFA09EHfClES36QOa5NvbZu%2F6IDfBVwiYywU%3D&reserved=0) (Table 3 shows the semseg results)
Code [https://github.com/NVIDIA/DL4AGX/tree/master/DEST](https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FNVIDIA%2FDL4AGX%2Ftree%2Fmaster%2FDEST&data=05%7C01%7Cboyinz%40nvidia.com%7Cbf078d69821449d1f4c908dab5e8c7da%7C43083d15727340c1b7db39efd9ccc17a%7C0%7C0%7C638022308636438546%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=9DLQZpEq1cN75%2FDf%2FniUOOUFS1ABX8FEUH02O6isGVQ%3D&reserved=0)
A webinar on its application [https://www.nvidia.com/en-us/on-demand/session/other2022-drivetraining/](https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.nvidia.com%2Fen-us%2Fon-demand%2Fsession%2Fother2022-drivetraining%2F&data=05%7C01%7Cboyinz%40nvidia.com%7Cbf078d69821449d1f4c908dab5e8c7da%7C43083d15727340c1b7db39efd9ccc17a%7C0%7C0%7C638022308636438546%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=8jrBC%2Bp3jGxiaW4vtSfhh6GozC3tRqGNjNoALM%2FOYxs%3D&reserved=0)

## Modification

Add backbone(smit.py) and head(dest_head.py) of DEST

## BC-breaking (Optional)

N/A

## Use cases (Optional)

N/A

### CODE REVIEW

**Comment by zhby99** [comment]
@MeowZheng Please help to review the PR. Thanks!
@lean76

**Comment by zhby99** [comment]
> @MeowZheng Please help to review the PR. Thanks! @lean76

**Comment by zhby99** [comment]
The checkpoints and logs for smit-b0~5 are under https://drive.google.com/drive/folders/1MOnW41XKPRYseOtxh_wgsbgpt7xwmeLb?usp=sharing for reference

**Comment by MeowZheng** [comment]
> @MeowZheng Please help to review the PR. Thanks! @lean76

Thanks for your contribution, please sign CLA first.

**Comment by zhby99** [comment]
> > @MeowZheng Please help to review the PR. Thanks! @lean76
> 
> Thanks for your contribution, please sign CLA first.

Thanks for your reply, I have signed CLA.

**Comment by MeowZheng** [comment]
Hi @zhby99 
To merge DEST quickly into MMSegmentaion repo, we suggest submitting it in community projects.
I'm sorry that I didn't mention this submission in the email, as we just started trying to use [`projects`](https://github.com/open-mmlab/mmsegmentation/tree/master/projects) for working with the community.

Please modify this pr following the [`example_project`](https://github.com/open-mmlab/mmsegmentation/tree/master/projects/example_project), and if you have any questions, please do not hesitate to contact me.

**Comment by zhby99** [comment]
@MeowZheng Hi Miao, Thanks for your suggestion, I have moved all related codes into projects directory, looking forward to your comments!

**Comment by MeowZheng** [comment]
> [CLA assistant check](https://cla-assistant.io/open-mmlab/mmsegmentation?pullRequest=2482) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/open-mmlab/mmsegmentation?pullRequest=2482) before we can accept your contribution.
> 
> **boyinz** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/open-mmlab/mmsegmentation?pullRequest=2482) it.

There might be 2 users in this pr **boyinz** and zhby99, and some commits are from **boyinz** who might be some default name of f your local git before you config it. CLA check is pending until you modify the commit history to remove **boyinz**.

Moreover, there are some ci problems, but we have fixed it in #2520 

Therefore, I suggest you might close this pr and create a new one based on the latest master.

Sorry to make your modification many times.

**Comment by zhby99** [comment]
@MeowZheng Thanks for your comments, I have rebased this pr on latest master and reset the commits with single user id.

**Review by MeowZheng — COMMENTED** [review]
please follow the [example project](https://github.com/open-mmlab/mmsegmentation/tree/master/projects/example_project#dummy-resnet-wrapper) to add `Prerequisites `and `Checklist`.

This project needs to provide the checkpoints, so would you like to share the model checkpoints？

**Comment by zhby99** [comment]
> please follow the [example project](https://github.com/open-mmlab/mmsegmentation/tree/master/projects/example_project#dummy-resnet-wrapper) to add `Prerequisites `and `Checklist`.
> 
> This project needs to provide the checkpoints, so would you like to share the model checkpoints？

Please find the checkpoints and logs under https://drive.google.com/drive/folders/1MOnW41XKPRYseOtxh_wgsbgpt7xwmeLb?usp=sharing . 
The prerequisites of this project is same with mmseg(i.e any environment that can build mmseg could train and test this project)
For checklist, I wonder if it is just something for reviewers and me to check? should we also add them to the README files?

**Comment by MeowZheng** [comment]
> > please follow the [example project](https://github.com/open-mmlab/mmsegmentation/tree/master/projects/example_project#dummy-resnet-wrapper) to add `Prerequisites `and `Checklist`.
> > This project needs to provide the checkpoints, so would you like to share the model checkpoints？
> 
> Please find the checkpoints and logs under https://drive.google.com/drive/folders/1MOnW41XKPRYseOtxh_wgsbgpt7xwmeLb?usp=sharing . The prerequisites of this project is same with mmseg(i.e any environment that can build mmseg could train and test this project) For checklist, I wonder if it is just something for reviewers and me to check? should we also add them to the README files?

As it may not be compatible between the different versions of mmseg and mmcv, we need to fix the version of them.
The checklist is for other users in the community, and they can understand the status of this algorithm.
Projects sometimes merge the model that is not official reimplement, just copy the model and model ckpt, whose performance can not be guaranteed, and the checklist works for providing the status of the model.

There is an example https://github.com/open-mmlab/mmsegmentation/tree/dev-1.x/projects/HieraSeg

BTW, I think we may talk more efficiently on WeChat if it is convenient for you. Would you like to email me your wechat id?

Thanks again for your contribution, and the documentation for community contribution processes needs to be improved

**Comment by zhby99** [comment]
> > > please follow the [example project](https://github.com/open-mmlab/mmsegmentation/tree/master/projects/example_project#dummy-resnet-wrapper) to add `Prerequisites `and `Checklist`.
> > > This project needs to provide the checkpoints, so would you like to share the model checkpoints？
> > 
> > 
> > Please find the checkpoints and logs under https://drive.google.com/drive/folders/1MOnW41XKPRYseOtxh_wgsbgpt7xwmeLb?usp=sharing . The prerequisites of this project is same with mmseg(i.e any environment that can build mmseg could train and test this project) For checklist, I wonder if it is just something for reviewers and me to check? should we also add them to the README files?
> 
> As it may not be compatible between the different versions of mmseg and mmcv, we need to fix the version of them. The checklist is for other users in the community, and they can understand the status of this algorithm. Projects sometimes merge the model that is not official reimplement, just copy the model and model ckpt, whose performance can not be guaranteed, and the checklist works for providing the status of the model.
> 
> There is an example https://github.com/open-mmlab/mmsegmentation/tree/dev-1.x/projects/HieraSeg
> 
> BTW, I think we may talk more efficiently on WeChat if it is convenient for you. Would you like to email me your wechat id?
> 
> Thanks again for your contribution, and the documentation for community contribution processes needs to be improved

Thanks for your suggestions. I have sent you an email with my wechat id.

**Comment by zhby99** [comment]
@MeowZheng Hi Miao, I have updated this PR with checklists and prerequisites

File: projects/dest/README.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+Transformer and its variants have shown state-of-the-art results in many vision tasks recently, ranging from image classification to dense prediction. Despite of their success, limited work has been reported on improving the model efficiency for deployment in latency-critical applications, such as autonomous driving and robotic navigation. In this paper, we aim at improving upon the existing transformers in vision, and propose a method for Dense Estimation with Simplified Transformer (DEST), which is efficient and particularly suitable for deployment on GPU-based platforms. Through strategic design choices, our model leads to significant reduction in model size, complexity, as well as inference latency, while achieving superior accuracy as compared to state-of-the-art in the task of self-supervised monocular depth estimation. We also show that our design generalize well to other dense prediction task such as semantic segmentation without bells and whistles.
+
+## Usage
+
+### Prerequisites
+
+- Python 3.7
+- PyTorch 1.6 or higher
+- [MMSegmentation](https://github.com/open-mmlab/mmsegmentation) v1.0.0rc2 or higher
+- mmcv v2.0.0rc4
```
Comment (MeowZheng) [inline projects/dest/README.md:16]: I think mmseg v1.0.0rc2 and mmcv v2.0.0rc4 maybe not right, please dump the environment by `conda env export > env.yml`.

**Review by MeowZheng — APPROVED** [review]

**Pull request merged**